src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheck.java

```diff
@@ -109,6 +111,19 @@
     public void beginTree(DetailAST rootAST) {
         currentImportControl = null;
         processCurrentFile = path.matcher(getFileContents().getFileName()).find();
+        unitName = null;
+
+        for (DetailAST sibling = rootAST; sibling != null; sibling = sibling.getNextSibling()) {
+            final int type = sibling.getType();
+
+            if (type == TokenTypes.CLASS_DEF
+                    || type == TokenTypes.ENUM_DEF
+                    || type == TokenTypes.INTERFACE_DEF
+                    || type == TokenTypes.ANNOTATION_DEF) {
+                unitName = sibling.findFirstToken(TokenTypes.IDENT).getText();
```

Comment:
as we come to unit term, we do not need to parse type name from file, lets just use name of the file.

Reply:
@romani If we do this, then the term unit name is wrong and it is file name. We specifically documented unit name is first compilation unit found. It is possible for file name and first unit name to be different. See http://checkstyle.sourceforge.net/config_misc.html#OuterTypeFilename .

Reply:
```
$ cat TestClass.java 
public class TestClass1 { }
$ javac TestClass.java 
TestClass.java:1: error: class TestClass1 is public, should be declared in a file named TestClass1.java
public class TestClass1 { }
       ^
1 error
```

> OuterTypeFilename .

We have this Check because it was not and error for all java compilers. So it become a Check. I think it is redundant right now on most recent compilers, but we suppose to work on any java file so Check is still present.

> If we are going with file name then unit should be changed to file.

File is too wide term - it ieverything, but compilation unit is file that is compile-able https://docs.oracle.com/javase/specs/jls/se7/html/jls-7.html#jls-7.3 , file name is compilation unit name is the same. Compilation unit contain whole content of java file. Our ImportControl do not care how much types are declared inside compilation unit, it checks only imports of compile-able file(compilation unit).

Reply:
> We have this Check because it was not and error for all java compilers.

It is not an error on maven as it will compile successfully files under `src/test/resources` when the first unit name isn't the same as the file name.
See 
https://github.com/checkstyle/checkstyle/blob/53d8e10fd3d049457f7aa3da503d48faef6264aa/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationAndroidStyle.java#L17
as one example of 64 I have found in our repository when scanning.

> File is too wide term - it ieverything

Since this is a `TreeWalker` check, it doesn't mean everything as `TreeWalker` makes sure only Java files are processed by it.
https://github.com/checkstyle/checkstyle/blob/53d8e10fd3d049457f7aa3da503d48faef6264aa/src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java#L93-L94

Reply:
> It is not an error on maven as it will compile successfully files

yes, but compilation unit is still a file with all classes inside it.
Looks like we need third opinion on this and make vote. I not completely against "file" term. 

> Since this is a TreeWalker check, it doesn't mean everything as TreeWalker makes sure only Java files are processed by it

I even did not think about TreeWalker, I just try to be user who see only configuration and try read and understand it without documentation.

Reply:
Changed to file.